lib/foreman_rh_cloud/engine.rb

```diff
@@ -210,6 +210,17 @@ def self.register_scheduled_task(task_class, cronline)
       )
     end
 
+    # Ideally this code belongs to an initializer. The problem is that Katello controllers are not initialized completely until after the end of the to_prepare blocks
+    # This means I can patch the controller only in the after_initialize block that is promised to run after the to_prepare
+    # initializer 'foreman_rh_cloud.allow_smart_proxy_actions', :before => :finisher_hook, :after => 'katello.register_plugin'  do |_app|
+    # end
+    config.after_initialize do
+      if defined?(Katello)
+        Katello::Api::V2::RepositoriesController.include Foreman::Controller::SmartProxyAuth
+        Katello::Api::V2::RepositoriesController.add_smart_proxy_filters(:index) # Add :features => proc { Foreman::Plugin.fact_importer_registry.fact_features }
```

Comment:
In this specific case we can expect to only have a single feature. Not sure what the name will be, but this plugin should also ensure it exists as part of a DB seed. Then it's sufficient to check for a single string instead of a proc from a registry.
```suggestion
        Katello::Api::V2::RepositoriesController.add_smart_proxy_filters(:index, features: 'My Feature')
```